ui/shared/components/panel/variants/Annotations.jsx

```diff
@@ -19,15 +19,28 @@ const LargeText = styled.div`
   font-size: 1.2em;
 `
 
-const UcscBrowserLink = ({ variant, useLiftover, includeEnd }) => {
-  const chrom = useLiftover ? variant.liftedOverChrom : variant.chrom
-  const pos = parseInt(useLiftover ? variant.liftedOverPos : variant.pos, 10)
+const UcscBrowserLink = ({ variant, useLiftover, includeEnd, isEnd }) => {
+  let { chrom, pos } = variant
+  const { rg37LocusEnd, end, endChrom } = variant
+  if (useLiftover) {
+    if (isEnd && rg37LocusEnd) {
+      chrom = rg37LocusEnd.contig
+      pos = rg37LocusEnd.position
+    } else {
+      chrom = variant.liftedOverChrom
+      pos = variant.liftedOverPos
+    }
+  } else if (isEnd) {
+    chrom = endChrom || chrom
+    pos = end
+  }
+  pos = parseInt(pos, 10)
   let genomeVersion = useLiftover ? variant.liftedOverGenomeVersion : variant.genomeVersion
   genomeVersion = genomeVersion === GENOME_VERSION_37 ? '19' : genomeVersion
 
   const highlight = variant.ref && `hg${genomeVersion}.chr${chrom}:${pos}-${pos + (variant.ref.length - 1)}`
   const highlightQ = highlight ? `highlight=${highlight}&` : ''
-  const endOffset = variant.end && variant.end - variant.pos
+  const endOffset = !endChrom && end && end - variant.pos
```

Comment:
I'm a little confused about this check. Maybe it would be better to check `includeEnd` here instead

Reply:
We can only compare the `end` and the `pos` when the `endChrom` is undefined. Otherwise, the `end` is for another chromosome. It doesn't make sense to compare them.
The `includeEnd` is checked when using the `endOffset` below. It can also be checked here but I think it also works without changing it.